core/src/haveibeenpwned.rs

```diff
@@ -0,0 +1,69 @@
+// check-if-email-exists
+// Copyright (C) 2018-2022 Reacher
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published
+// by the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use crate::util::constants::LOG_TARGET;
+use pwned::api::PwnedBuilder;
+
+use serde::{Deserialize, Serialize};
+
+/// The result of the check_haveibeenpwned function.
+#[derive(Debug, Deserialize, Serialize)]
+pub struct CheckHaveIBeenPwnedDetails {
+	/// The number of times the email has been found.
+	pub breach_found_count: Option<usize>,
+	/// Error message if any.
+	pub error: Option<String>,
+}
+
+/// Check if the email has been found in any breach or paste using the
+/// HaveIBeenPwned API.
+/// This function will return the number of times the email has been found in
+/// any breach.
+pub async fn check_haveibeenpwned(
+	to_email: &str,
+	api_key: &Option<String>,
```

Comment:
Can we do 
```suggestion
	api_key: Option<String>,
```
here?

And then, we don't need to clone on L41